bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -0,0 +1,286 @@
+        plasmod_options["params"].set_parameter("kappa", kappa, "dimensionless", source)
+        plasmod_options["params"].set_parameter("delta", delta, "dimensionless", source)
+        plasmod_options["params"].set_parameter(
+            "kappa_95", kappa_95, "dimensionless", source
+        )
+        plasmod_options["params"].set_parameter(
+            "delta_95", delta_95, "dimensionless", source
+        )
+
+        plasmod_solver = PlasmodTransportSolver(
+            params=plasmod_options["params"],
+            build_config=plasmod_options["build_config"],
+        )
+        plasmod_solver.execute()
+
+        if plot:
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.pprime(plasmod_solver.x), "pprime", "-"
+            )
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.ffprime(plasmod_solver.x), "ffrime", "-"
+            )
+
+        # generate mesh for the Grad-Shafranov solver
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_coarse,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {
+            "lcar": lcar_coarse,
+            "physical_group": "plasma_face",
+        }
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=verbose)
+
+        mesh, boundaries, subdomains, labels = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )
+
+        # initialize the Grad-Shafranov solver
+        p_order = gs_options["p_order"]
+        gs_solver = FemGradShafranovFixedBoundary(mesh, p_order=p_order)
+
+        bluemira_print("Solving fixed boundary Grad-Shafranov...")
+
+        gs_solver.solve(
+            plasmod_solver.pprime,
+            plasmod_solver.ffprime,
+            plasmod_solver.I_p,
+            tol=gs_options["tol"],
+            max_iter=gs_options["max_iter"],
+            relaxation=gs_relaxation,
+            verbose_plot=gs_options["verbose_plot"],
+        )
+
+        # create the finer mesh to calculate the isofluxes
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_fine,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {"lcar": lcar_fine, "physical_group": "plasma_face"}
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=verbose)
+
+        mesh, boundaries, subdomains, labels = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )
+
+        points = mesh.coordinates()
+        x2d_data = np.array([gs_solver.psi_norm_2d(x) for x in points])
+
+        # calculate kappa_95 and delta_95
+        r_geo, kappa_95, delta_95 = calculate_plasma_shape_params(
+            points, x2d_data, [np.sqrt(0.95)]
+        )
+        r_geo, kappa_95, delta_95 = r_geo[0], kappa_95[0], delta_95[0]
```

Comment:
As discussed, there is an alternative way to get the shape parameters of the 95th percentile flux surface: by optimisation (grid-searching) of the smooth psi_norm2d function. 

I attempted an implementation of this (using scipy's minimize, which is useful for sketching out optimisation problems) and the results were rather strange.
First, it seems like I can recover the etrema of the 95th flux surface very well, and get very similar results to the mesh approach. I even believe that if you were to use finer and finer mesh size, the extrema would tend towards those of the optimisation approach which for comparably speaking is instantaneous.

However, this change completely breaks the PLASMOD <-> fixed boundary convergence, with kappa and delta values reaching non-sensical values.

On paper, the convergence is improved:

![image](https://user-images.githubusercontent.com/26097289/178464378-d7ae5d51-9a96-4ca6-abf5-129e3207a7b1.png)

but does not fill me with joy as I had hoped, given the resulting values of kappa and delta:
* kappa_u_final = 2.14 (!!)
* delta_u_final = 0.878 (!!)

v.s for the mesh approach:
* kappa_u_final = 1.87
* delta_u_final=0.4772

One thing to note is that changing the p_order to 1 improves the discrepancy between the opt mesh approach, but still results in crazy results for the opt approach.

This requires further investigation, as I believe it has the potential to remove one meshing step, and improve convergence. At present I simply don't understand why the result is nonsense, sadly.

Reply:
OK, the problem was down to a simple `np.sqrt(0.95)` vs `0.95`..!

Now it works, and appears more stable and faster across the board:
![image](https://user-images.githubusercontent.com/26097289/178730884-02b9c0c8-01b4-4573-9199-054900fc79e0.png)
with the results being approximately equal in terms of kappa and delta.

Reply:
OK, done now